django_cloud_deploy/cli/prompt.py

```diff
@@ -541,50 +543,102 @@ class GoogleExistingProjectId(TemplatePrompt):
+    def _validate(self, backend: str, active_account: str, project_id: str):
         """Validates that a string is a valid project id.
 
         Args:
-            s: The string to validate.
+            backend: The backend that will be used to host the app.
+            active_account: Account that is logged in on gcloud cli.
+            project_id: Id of the Google Project.
 
         Raises:
             ValueError: if the input string is not valid.
         """
 
-        if not re.match(r'[a-z][a-z0-9\-]{5,29}', s):
-            raise ValueError(('Invalid Google Cloud Platform Project ID "{}": '
-                              'must be between 6 and 30 characters and contain '
-                              'lowercase letters, digits or hyphens').format(s))
+        if not re.match(r'[a-z][a-z0-9\-]{5,29}', project_id):
+            raise ValueError(
+                ('Invalid Google Cloud Platform Project ID "{}": '
+                 'must be between 6 and 30 characters and contain '
+                 'lowercase letters, digits or hyphens').format(project_id))
+
+        if not self.project_client.project_exists(project_id):
+            raise ValueError('Project {} does not exist'.format(project_id))
+
+        if not self._has_correct_permissions(backend, project_id,
+                                             active_account):
+            msg = 'User has incorrect permissions to deploy.'
+            if backend == 'gae':
+                msg = 'User must be a Project Owner to deploy on GAE'
+            elif backend == 'gke':
+                msg = ('User does not have correct permissions'
+                       'to deploy on GKE')
+            raise ValueError(msg)
+
+    def _has_correct_permissions(self, backend: str, project_id: str,
+                                 active_account: str):
+        """Validates that the user has the permissions to deploy onto project.
 
-        if not self.project_client.project_exists(s):
-            raise ValueError('Project {} does not exist'.format(s))
+        Args:
+            project_id: Id of the existing project.
+            backend: The backend that will be used to host the app.
+            active_account: Account that is logged in on gcloud cli.
+        """
+        # The user must have logged in
+        assert active_account != ''
+
+        permissions = self.project_client.get_project_permissions(project_id)
+        owner_permission = list(
+            filter(lambda d: d.get('role') == 'roles/owner', permissions))
+        editor_permission = list(
+            filter(lambda d: d.get('role') == 'roles/editor', permissions))
+
+        owners = []
+        editors = []
+        if owner_permission:
+            owners = owner_permission[0].get('members', [])
+        if editor_permission:
+            editors = editor_permission[0].get('members', [])
+
+        active_account = re.escape(active_account)
+        if re.search(active_account, ' '.join(owners)):
```

Comment:
For example, my account is "tim@gmail.com", and this account does not exist in either owners or editors. And there is another account called "blablatim@gmail.com", who is the owner. Then re.search will return true when active_account is "tim@gmail.com", but it is not necessarily an owner.

Reply:
that is a good catch.